### PR TITLE
 `sendpay` and `pay` return detailed failure

### PR DESCRIPTION
**Pull request by ZmnSCPxj**
Fixes: #866

Based on #903

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
Rebased as #903 is merged in.

**Comment by ZmnSCPxj** [comment]
work-in-progress; I will change the error codes as per this comment: https://github.com/ElementsProject/lightning/issues/866#issuecomment-362882530

**Comment by ZmnSCPxj** [comment]
Ready for review

File: doc/lightning-pay.7.txt [hunk 1]
```diff
@@ -30,7 +30,42 @@ RETURN VALUE
...
+  the error will be routing failure object.
+* 204. Failure along route; retry a different route. The 'data'
+  field of the error will be routing failure object.
+* 205. Unable to find a route.
+
+A routing failure object has the fields below:
+
+* 'origin_index'. The index of the node along the route that
+  reported the error. 0 for the first hop node, -1 if the local
+  node reported the error.
```
Comment (rustyrussell) [inline doc/lightning-pay.7.txt]: Let's add one here, which is far clearer?  eg:

    'origin_index'. The index of the node along the route that reported the error: 0 for the local node,
    1 for the first hop.

**Comment by rustyrussell** [comment]
Tiny JSON API change requested, rest looks good.

File: doc/lightning-pay.7.txt (hunk 1 above)
Comment (ZmnSCPxj) [reply doc/lightning-pay.7.txt]: If you do not mind, also change s/origin_index/erring_index/, then? Internal origin_index counts first hop as 0, I simply extended this to -1 for local node.  Better to reduce confusion for later devs: if they see `origin_index` as return value of `pay`/`sendpay` and then assume that internal `origin_index` created by `unwrap_onionreply` is the same, when in fact the RPC API adds one to the internal value.

Comment (rustyrussell) [reply doc/lightning-pay.7.txt]: Agreed, good idea. That is a better name!

**Comment by ZmnSCPxj** [comment]
Changed as per review, CI passed, ready for merge.

**Comment by cdecker** [comment]
Excellent work @ZmnSCPxj 
ACK f20a15a

**Pull request merged**